libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java

```diff
@@ -90,7 +90,7 @@ private void extractStepResources(RouteProgress progress) {
 
   private void formatStepDistance(Context context, RouteProgress progress,
                                   Locale locale, @NavigationUnitType.UnitType int unitType) {
-    if (distanceUtils == null || this.locale != locale || this.unitType != unitType) {
+    if (distanceUtils == null || this.locale.equals(locale) || this.unitType != unitType) {
```

Comment:
To match previous behavior (`this.locale != locale`), shouldn't be `!this.locale.equals(locale)` instead? Maybe it's a bug fix that I'm missing. Could you clarify?